### PR TITLE
Keyboard shortcuts

### LINKED ISSUE #104: Add keyboard shortcuts
The site could do with some keyboard shortcuts for faster navigation of the site.

I was thinking something like this:
- `gt` or `gh` Timeline
- `gi` Inbox
- `gd` Discover
- `gp` Profile
- `j`/`k` for changing the focused question (down/up)
- `c` Write a comment
- `l`? Smile/unsmile
- `a` Answers
- `n` Ask a question

### PR DESCRIPTION
**Pull request by raccube**
Resolves #104

### CODE REVIEW

File: app/views/inbox/show.html.haml [hunk 1]
```diff
@@ -11,4 +11,5 @@
       class: "btn btn-light",
       method: :get,
       params: { last_id: @inbox_last_id, author: @author }.compact,
+      data: { hotkey: "." },
```
Comment (nilsding) [inline app/views/inbox/show.html.haml]: this only seems to work once

**Review by nilsding — APPROVED** [review]
overall nice!

File: app/views/inbox/show.html.haml (hunk 1 above)
Comment (raccube) [reply app/views/inbox/show.html.haml]: Have added a Stimulus controller to deal with this.

File: app/javascript/retrospring/initializers/hotkey.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import { install } from '@github/hotkey'
+
+export default function (): void {
+  document.addEventListener('turbo:load', () => {
+    console.debug("Init Hotkey.")
```
Comment (nilsding) [inline app/javascript/retrospring/initializers/hotkey.ts]: i think @pixeldesu will have something to say about this ;-)

**Review by nilsding — APPROVED** [review]

File: app/javascript/retrospring/initializers/hotkey.ts (hunk 2 above)
Comment (pixeldesu) [inline app/javascript/retrospring/initializers/hotkey.ts]: Remove this debug statement, please!

**Review by pixeldesu — CHANGES_REQUESTED** [review]
Looks and works fine generally, just missing the hotkey dialog actually being implemented!

**Comment by raccube** [comment]
@pixeldesu that's weird, the dialog opens fine for me.

**Comment by pixeldesu** [comment]
@raccube I don't see a link/button anywhere in the UI for it, and in Firefox pressing `?` just opens the browsers Quick Find menu.

**Comment by raccube** [comment]
> @raccube I don't see a link/button anywhere in the UI for it, and in Firefox pressing `?` just opens the browsers Quick Find menu.

image

**Comment by nilsding** [comment]
@pixeldesu could this be a kezboard lazout/лунищфкв дфнщге thing?

**Comment by raccube** [comment]
Looks like it. I could add `Shift+ß` as a shortcut to account for it.

File: app/views/navigation/dropdown/_profile.html.haml [hunk 3]
```diff
@@ -34,6 +34,10 @@
       %i.fa.fa-fw.fa-flask
       = t(".feedback.features")
     .dropdown-divider
+    %a.dropdown-item{ href: "#", data: { bs_target: "#modal-hotkeys", bs_toggle: "modal", hotkey: "Shift+?,?" } }
+      %i.fa.fa-keyboard
+      = t(".hotkeys")
+    .dropdown-divider
```
Comment (pixeldesu) [inline app/views/navigation/dropdown/_profile.html.haml]: This section of the menu is inside the `if APP_CONFIG["canny"]` condition, please move that outside of that.

**Review by pixeldesu — CHANGES_REQUESTED** [review]

File: app/views/navigation/dropdown/_profile.html.haml (hunk 3 above)
Comment (raccube) [reply app/views/navigation/dropdown/_profile.html.haml]: Oops.

**Review by pixeldesu — APPROVED** [review]

**Pull request merged**